jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/JibExtension.java

```diff
@@ -97,19 +100,34 @@ public void container(Action<? super ContainerParameters> action) {
     action.execute(container);
   }
 
-  public void extraDirectory(Action<? super ExtraDirectoryParameters> action) {
-    action.execute(extraDirectory);
+  @Deprecated
+  public void extraDirectory(Action<? super ExtraDirectoriesParameters> action) {
+    extraDirectoryConfigured = true;
+    action.execute(extraDirectories);
+  }
+
+  public void extraDirectories(Action<? super ExtraDirectoriesParameters> action) {
+    extraDirectoriesConfigured = true;
+    action.execute(extraDirectories);
+  }
+
+  @Deprecated
+  // for the deprecated "jib.extraDirectory" config parameter
+  public void setExtraDirectory(File extraDirectory) {
+    extraDirectoryConfigured = true;
+    this.extraDirectories.setPath(extraDirectory);
   }
 
   /**
    * Sets extra directory paths. {@code extraDirectories} can be any suitable object describing file
-   * paths convertible by {@link Project#files} (such as {@code List<File>}).
+   * paths convertible by {@link Project#files} (such as {@link File}, {@code List<File>} or {@code
+   * List<String>}).
    *
    * @param extraDirectories paths to set.
    */
-  // non-plural to retain backward-compatibility for the "jib.extraDirectory" config parameter
-  public void setExtraDirectory(Object extraDirectories) {
-    this.extraDirectory.setPath(extraDirectories);
+  public void setExtraDirectories(Object extraDirectories) {
```

Comment:
Wouldn't we want to avoid adding getters and setters for extraDirectories? I thought in maven we're pushing everyone to use the complex type configuration for `<extraDirectories><paths.....` I would image we would want to do that here as well... and just not allow `jib.extraDirectories = asdf`

Reply:
Sounds good with being consistent. We allowed the shortcut to remain backward compatibility after introducing `.permissions`, but we can clean that up since we are making breaking changes with deprecation.